src/utils/CrossChainBridge.ts

```diff
@@ -177,3 +200,7 @@ export function isETHBridge(address: string): boolean {
 export function isBSCBridge(address: string): boolean {
   return address.toLowerCase() == bridgeBSC.toLowerCase()
 }
+
+function isSignatureFederation(address: string): boolean {
+  return address.toLowerCase() == signaturesFederationETH.toLowerCase() || address.toLowerCase() == signaturesFederationBSC.toLowerCase()
```

Comment:
This could be made more extensible by using a set for signature federations, and then using a getter

Reply:
you mean a typed map?

Reply:
A typed map or a Set. Like:

``const federatorContracts = new Set<string>
federatorContracts.add(signaturesFederationETH.toLowerCase())
...
if(federatorContracts.has(address)) {
  code
}``

The map could also maybe be set in the code so that it doesn't have to be hard-coded, I'm not sure

Reply:
tbh I don't how this makes it move extensible

Reply:
It's a bit faster and avoids a long chain of if/or, but it's probably just a preference thing. The set could also be dynamically changed in the code I think, so that the federator addresses don't need to be hard-coded.

On the first event with a signature from the signature federation, the event address can be added to the set. The any further federations don't have to be manually added

Reply:
I don't think you can add something to a set on one event handler that would exist on a later event handler (unless you save it to the store)